### PR TITLE
Port to angular

### PR DESCRIPTION
**Pull request by yusra-haider**
see ebi-ait/dcp-ingest-central#173

### CODE REVIEW

**Review by amnonkhen — CHANGES_REQUESTED** [review]

**Comment by MightyAx** [comment]
Does this PR need to merge to `dev` instead of `master`?

**Comment by jacobwindsor** [comment]
@MightyAx There's no dev branch for the catalog yet. I suggest we make one after we do this body of work

File: package.json [hunk 1]
```diff
@@ -1,44 +1,59 @@
...
+    "@angular/core": "~9.1.1",
+    "@angular/forms": "~9.1.1",
+    "@angular/platform-browser": "~9.1.1",
+    "@angular/platform-browser-dynamic": "~9.1.1",
+    "@angular/router": "~9.1.1",
+    "rxjs": "~6.5.4",
+    "tslib": "^1.10.0",
+    "zone.js": "~0.10.2"
   },
   "husky": {
```
Comment (amnonkhen) [inline package.json:54]: Do we need this husky element?

File: src/app/app.component.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css'],
+})
+export class AppComponent {
+  title = 'projects-index';
```
Comment (amnonkhen) [inline src/app/app.component.ts:9]: Do we use the title variable?

Comment (yusra-haider) [reply src/app/app.component.ts:9]: No, it was autogenerated by angular

File: package.json (hunk 1 above)
Comment (yusra-haider) [reply package.json:54]: @jacobwindsor ?

File: projects-index/src/index.html [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  type="text/css"
+  media="all"
+/>
+
+<body class="vf-body">
+  <app-root></app-root>
+</body>
+
+<!-- embl-ebi global footer -->
+<link
```
[the remark this answers is not in the document]
Comment (yusra-haider) [reply projects-index/src/index.html]: I did, but it didn't work as expected

File: projects-index/src/assets/scraper.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    parser.add_argument("-u", "--url", help="Base URL for Ingest API", default="https://api.ingest.archive.data.humancellatlas.org")
+
+    args = parser.parse_args()
+
+    uuids = get_uuids(args.input)
+
+    if args.clean:
+        print("Running with clean option. This will update preexisting timestamps.")
+
+    with open(args.output, 'r+') as out:
```
[the remark this answers is not in the document]
Comment (yusra-haider) [reply projects-index/src/assets/scraper.py]: deleted this file as it's no longer needed after the changes to core

File: projects-index/src/assets/scraper.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+            new_contributors.append(contributor)
+        except KeyError:
+            new_contributors.append(contributor)
+    return new_contributors
+                
+
+def make_dcp_link(prj_uuid):
+    # TODO This should probably be replicated in the ingest API once we move there
+    catalog = "dcp2"
+    azul_proj_url = "https://service.azul.data.humancellatlas.org/index/projects/{}?catalog={}".format(prj_uuid, catalog)
```
[the remark this answers is not in the document]
Comment (yusra-haider) [reply projects-index/src/assets/scraper.py]: deleted this file as it's no longer needed after integration with core endpoint

File: projects-index/package.json [hunk 6]
```diff
@@ -0,0 +1,47 @@
+{
+  "name": "projects-index",
+  "version": "0.0.0",
```
[the remark this answers is not in the document]
Comment (yusra-haider) [reply projects-index/package.json]: what version are we keeping?

cc: @jacobwindsor

[the remark this answers is not in the document]
Comment (yusra-haider) [reply projects-index/package.json]: keeping at as `0.1.0`

File: projects-index/package.json [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  "scripts": {
+    "ng": "ng",
+    "start": "ng serve",
+    "build": "ng build",
+    "test": "ng test",
+    "lint": "ng lint",
+    "e2e": "ng e2e"
+  },
+  "private": true,
+  "dependencies": {
```
[the remark this answers is not in the document]
Comment (yusra-haider) [reply projects-index/package.json]: it was added as a pre-built CDN
https://stable.visual-framework.dev/

**Review by MightyAx — APPROVED** [review]
Ran locally and it looks good to me.

File: src/app/app.component.ts (hunk 2 above)
Comment (aaclan-ebi) [reply src/app/app.component.ts:9]: I think it was only being used in the example HTML page.

File: src/environments/environment.dev.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
+export const environment = {
+  production: true,
```
Comment (aaclan-ebi) [inline src/environments/environment.dev.ts:2]: is this really true?

File: src/app/app.component.ts (hunk 2 above)
Comment (yusra-haider) [reply src/app/app.component.ts:9]: shall I remove it?

Comment (yusra-haider) [reply src/app/app.component.ts]: removed

File: angular.json [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+          "builder": "@angular-devkit/build-angular:browser",
+          "options": {
+            "outputPath": "dist/projects-index",
+            "index": "src/index.html",
+            "main": "src/main.ts",
+            "polyfills": "src/polyfills.ts",
+            "tsConfig": "tsconfig.app.json",
+            "aot": true,
+            "assets": [
+              "src/favicon.ico",
```
Comment (aaclan-ebi) [inline angular.json:23]: should we just remove this line, favicon.ico is missing

File: projects-index/src/index.html [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+  rel="import"
+  href="https://www.embl.org/api/v1/pattern.html?filter-content-type=article&filter-id=106902&pattern=node-body&source=contenthub"
+  data-target="self"
+  data-embl-js-content-hub-loader
+/>
+<script
+  defer
+  src="https://assets.emblstatic.net/vf/v2.4.5/scripts/scripts.js"
```
[the remark this answers is not in the document]
Comment (yusra-haider) [reply projects-index/src/index.html]: I haven't looked at this yet, will do in a follow up PR

File: src/app/projects/projects.service.ts [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+        uuid: obj.uuid.uuid,
+        dcpUrl: obj.dcp_url,
+        addedToIndex: obj.cataloguedDate,
+        date: obj.cataloguedDate ? this.formatDate(obj.cataloguedDate) : '-',
+        title: obj.content.project_core.project_title,
+        organs:
+          obj.organ?.ontologies?.map((organ) => organ.ontology_label) ?? [],
+        technologies:
+          obj.technology?.ontologies?.map((tech) => tech.ontology_label) ?? [],
+        // Temp fix until ena accessions fixed in core
```
Comment (aaclan-ebi) [inline src/app/projects/projects.service.ts:46]: please create an ops ticket! thanks!

**Review by aaclan-ebi — APPROVED** [review]
LGTM!

**Review by amnonkhen — APPROVED** [review]
Looks awesome for now!
Thanks for the hard work.

File: src/app/projects/projects.service.ts (hunk 11 above)
Comment (yusra-haider) [reply src/app/projects/projects.service.ts:46]: https://app.zenhub.com/workspaces/operations-5fa2d8f2df78bb000f7fb2b5/issues/ebi-ait/hca-ebi-wrangler-central/345

**Pull request merged**